z3c/dependencychecker/dotted_name.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+import hashlib
+from functools import total_ordering
+
+
+@total_ordering
+class DottedName(object):
+
+    def __init__(
+        self,
+        name,
+        file_path=None,
+        line_number=None,
+        is_test=False,
+    ):
+        self.name = name
+        self.safe_name = self._get_safe_name(name)
+        self.namespaces = self._get_namespaces(self.safe_name)
+        self.is_namespaced = self._is_namespaced(self.namespaces)
+        self.file_path = file_path
+        self.line_number = line_number
+        self.is_test = is_test
+
+    @classmethod
+    def from_requirement(cls, requirement, file_path=None):
+        """A requirement in this method's context is a
+        pkg_resources.Requirement
+        """
+        return cls(
+            requirement.project_name,
+            file_path=file_path,
+        )
+
+    @staticmethod
+    def _get_safe_name(name):
+        safe_name = name.lower().replace('-', '_')
+        return safe_name
+
+    @staticmethod
+    def _get_namespaces(safe_name):
+        parts = safe_name.split('.')
+        return parts
+
+    @staticmethod
+    def _is_namespaced(namespaces):
+        return bool(len(namespaces) - 1)
+
+    def __lt__(self, other):
+        if not isinstance(other, DottedName):
+            return NotImplemented
+        return self.name < other.name
+
+    def __eq__(self, other):
+        if not isinstance(other, DottedName):
+            return NotImplemented
+
+        return self.safe_name == other.safe_name
+
+    def __repr__(self):
+        return '<DottedName {0}>'.format(self.name)
+
+    def __hash__(self):
+        digest = hashlib.sha256(self.safe_name).hexdigest()
+        return int(digest, 16) % 10**8
+
+    def __contains__(self, item):
```

Comment:
This is the only place where I'd like a short docstring. What does containment mean here (or rather: how do you want to use it).

Looks like you're making something that is nicely pythonic, btw! Equality, hash, etc. Good job!

Reply:
There is a database class that actually does the heavy lifting, and there I use sets and list comprehensions to filter, so having all these dunder methods really keeps code simpler.

I will write a docstring on that one.